File: src/main/java/com/sem/keeper/service/SemAuthSuccessHandler.java
package com.sem.keeper.service;

import com.sem.keeper.entity.UserEntity;
import com.sem.keeper.repo.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.Authentication;
import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
import org.springframework.stereotype.Component;
import org.springframework.stereotype.Service;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpSession;
import java.io.IOException;

@Component
public class SemAuthSuccessHandler implements AuthenticationSuccessHandler {

    @Autowired
    UserService userRepo;

    @Override
    public void onAuthenticationSuccess(HttpServletRequest request, HttpServletResponse response, Authentication authentication) throws IOException, ServletException {
        String name = authentication.getName();
        System.out.println(name);
        System.out.println(userRepo);
        UserEntity user = userRepo.findByEmail(name);

        HttpSession session = request.getSession();
        session.setAttribute("user",user);

        response.setStatus(HttpServletResponse.SC_OK);
        response.sendRedirect("/");
    }
}

File: src/main/java/com/sem/keeper/service/SemAuthSuccessHandler.kt
package com.sem.keeper.service

import com.sem.keeper.repo.UserRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.Authentication
import org.springframework.security.web.authentication.AuthenticationSuccessHandler
import org.springframework.stereotype.Component
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@Component
class SemAuthSuccessHandler : AuthenticationSuccessHandler {

    @field:Autowired
    private lateinit var userRepository: UserRepository

    override fun onAuthenticationSuccess(
        request: HttpServletRequest,
        response: HttpServletResponse,
        authentication: Authentication
    ) {
        val name = authentication.name
        val user = userRepository.findByEmail(name)
        val session = request.session
        session.setAttribute("user", user)
        response.status = HttpServletResponse.SC_OK
        response.sendRedirect("/")
    }
}